src/analysis/processing/qgsalgorithmnormalraster.cpp

```diff
@@ -0,0 +1,187 @@
+}
+
+bool QgsNormalRasterAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( feedback )
+  mCrs = parameterAsCrs( parameters, QStringLiteral( "TARGET_CRS" ), context );
+  mExtent = parameterAsExtent( parameters, QStringLiteral( "EXTENT" ), context, mCrs );
+  mPixelSize = parameterAsDouble( parameters, QStringLiteral( "PIXEL_SIZE" ), context );
+  mTypeId = parameterAsInt( parameters, QStringLiteral( "OUTPUT_TYPE" ), context );
+
+  if ( mTypeId == 0 )
+  {
+    mRasterDataType = Qgis::Float32;
+  }
+  else
+  {
+    mRasterDataType = Qgis::Float64;
+  }
+
+  mMean = parameterAsDouble( parameters, QStringLiteral( "MEAN" ), context );
+  mStdev = parameterAsDouble( parameters, QStringLiteral( "STDEV" ), context );
+
+  mNormalDoubleDistribution = std::normal_distribution<double>( mMean, mStdev );
+
+  return true;
+}
+
+QVariantMap QgsNormalRasterAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  const QString outputFile = parameterAsOutputLayer( parameters, QStringLiteral( "OUTPUT" ), context );
+  QFileInfo fi( outputFile );
+  const QString outputFormat = QgsRasterFileWriter::driverForExtension( fi.suffix() );
+
+  int rows = std::max( std::ceil( mExtent.height() / mPixelSize ), 1.0 );
+  int cols = std::max( std::ceil( mExtent.width() / mPixelSize ), 1.0 );
+
+  //build new raster extent based on number of columns and cellsize
+  //this prevents output cellsize being calculated too small
+  QgsRectangle rasterExtent = QgsRectangle( mExtent.xMinimum(), mExtent.yMaximum() - ( rows * mPixelSize ), mExtent.xMinimum() + ( cols * mPixelSize ), mExtent.yMaximum() );
+
+  std::unique_ptr< QgsRasterFileWriter > writer = qgis::make_unique< QgsRasterFileWriter >( outputFile );
+  writer->setOutputProviderKey( QStringLiteral( "gdal" ) );
+  writer->setOutputFormat( outputFormat );
+  std::unique_ptr<QgsRasterDataProvider > provider( writer->createOneBandRaster( mRasterDataType, cols, rows, rasterExtent, mCrs ) );
+  if ( !provider )
+    throw QgsProcessingException( QObject::tr( "Could not create raster output: %1" ).arg( outputFile ) );
+  if ( !provider->isValid() )
+    throw QgsProcessingException( QObject::tr( "Could not create raster output %1: %2" ).arg( outputFile, provider->error().message( QgsErrorMessage::Text ) ) );
+
+  double step = rows > 0 ? 100.0 / rows : 1;
+
+  for ( int i = 0; i < rows ; i++ )
+  {
+    if ( feedback->isCanceled() )
+    {
+      break;
+    }
+    //prepare raw data depending on raster data type
+    QgsRasterBlock block( mRasterDataType, cols, 1 );
+    switch ( mTypeId )
+    {
+      case ( 0 ):
+      {
+        std::vector<float> float32Row( cols );
+        for ( int i = 0; i < cols; i++ )
+        {
+          float32Row[i] = static_cast<float>( mNormalDoubleDistribution( mRandomDevice ) );
```

Comment:
I don't think this is quite correct use of the random classes. I think you should be using mMersenneTwister here, having first seeded mMersenneTwister with mRandomDevice...

Reply:
See the example from https://en.cppreference.com/w/cpp/numeric/random/normal_distribution:

```
std::random_device rd{};
    std::mt19937 gen{rd()};
 
    // values near the mean are the most likely
    // standard deviation affects the dispersion of generated values from the mean
    std::normal_distribution<> d{5,2};
 
    std::map<int, int> hist{};
    for(int n=0; n<10000; ++n) {
        ++hist[std::round(d(gen))];
    }
```

Reply:
Ah yes, I go this mixed up with all the abbreviations in the example code ;)

Reply:
fixed